Userland/Applications/FileManager/FileUtils.cpp

```diff
@@ -67,4 +68,91 @@ void delete_paths(const Vector<String>& paths, bool should_confirm, GUI::Window*
     }
 }
 
+void create_trash_bin()
+{
+    auto trash_bin_path = String::formatted("{}/.trash", getenv("HOME"));
+    auto trash_bin_files = String::formatted("{}/files", trash_bin_path);
+    auto trash_bin_info = String::formatted("{}/info", trash_bin_path);
+
+    if (Core::File::is_directory(trash_bin_path)) {
+        dbgln("Trash directory exists.");
+    } else {
+        if (mkdir(trash_bin_path.characters(), 0755) < 0) {
+            auto saved_errno = errno;
+            warnln("Failed to create directory {}: {}", trash_bin_path, strerror(saved_errno));
+        }
+    }
+
+    if (!Core::File::is_directory(trash_bin_files)) {
+        if (mkdir(trash_bin_files.characters(), 0755) < 0) {
+            auto saved_errno = errno;
+            warnln("Failed to create directory {}: {}", trash_bin_files, strerror(saved_errno));
+        }
+    }
+
+    if (!Core::File::is_directory(trash_bin_info)) {
+        if (mkdir(trash_bin_info.characters(), 0755) < 0) {
+            auto saved_errno = errno;
+            warnln("Failed to create directory {}: {}", trash_bin_info, strerror(saved_errno));
+        }
+    }
+}
+
+[[maybe_unused]] void move_to_trash(String const& path)
+{
+    [[maybe_unused]] auto file_info = "[Trash Info]\nPath={}\nDeletionDate={}\n";
+
+    auto file_name = LexicalPath(path).basename();
+    auto file_info_name = String::formatted("{}.trashinfo", file_name);
+    auto deletion_date = Core::DateTime::now().to_string("%Y-%m-%dT%H:%M:%S");
+
+    auto trash_bin_path = String::formatted("{}/.trash", getenv("HOME"));
+    auto trash_bin_files = String::formatted("{}/files", trash_bin_path);
+    auto trash_bin_info = String::formatted("{}/info", trash_bin_path);
+
+    create_trash_bin();
+
+    auto trash_into_path = String::formatted("{}/{}", trash_bin_info, file_info_name);
+
+    auto trashed_file_info = String::formatted(file_info, path, deletion_date);
+
+    auto file_or_error = Core::File::open(trash_into_path, Core::IODevice::WriteOnly);
+    if (file_or_error.is_error())
+        return;
```

Comment:
This function prints debug output upon failure in most (all?) other instances. Why not here too?